dbms/src/Interpreters/Users.cpp

```diff
@@ -69,15 +70,25 @@ class IPAddressPattern : public IAddressPattern
         else
         {
             String addr(str, 0, pos - str.c_str());
-            UInt8 prefix_bits_ = parse<UInt8>(pos + 1);
-
+            String str_mask(str, addr.length() + 1, str.length() - addr.length() - 1);
+            UInt8 prefix_bits_;
+            if (isDigits(str_mask))
+            {
+                prefix_bits_ = parse<UInt8>(pos + 1);
+            }
+            else
+            {
+                Poco::Net::IPAddress ip_mask(str_mask);
+                prefix_bits_ = ip_mask.prefixLength();
```

Comment:
The goal or your task is to add support for "mark with holes" like
`255.255.0.255`
or
`ffff:ffff:ff00:0:ffff:ffff::`
(as in example `2a02:6b8:c00::660:0:0/ffff:ffff:ff00:0:ffff:ffff::`)

it's not just a prefix.